### PR TITLE
feat(object store): add LoggingLayer for OpenDAL storage

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
Great, thanks!

File: src/object_store/src/object/opendal_engine/azblob.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ impl OpendalObjectStore {
         builder.account_key(&account_key);
         let op: Operator = Operator::new(builder)?
             .layer(RetryLayer::default())
+            .layer(LoggingLayer::default())
```
Comment (Xuanwo) [inline src/object_store/src/object/opendal_engine/azblob.rs:41]: It's better to implement something like https://opendal.apache.org/docs/rust/opendal/struct.OperatorBuilder.html#examples

But this PR is Ok to go, we can refactor this part later.

**Review by hzxa21 — APPROVED** [review]

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -370,9 +371,8 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
 
         layers.push(Box::new(MetricsLayer::new(registry).with_filter(filter)));
     }
-
-    tracing_subscriber::registry().with(layers).init();
-
+    LogTracer::init().expect("Failed to set logger");
+    tracing_subscriber::registry().with(layers).set_default();
```
Comment (wcy-fdu) [inline src/utils/runtime/src/lib.rs:375]: Not sure whether can replace `tracing_subscriber::registry().with(layers).init();` with `tracing_subscriber::registry().with(layers).set_default();`🤔

Comment (xxchan) [reply src/utils/runtime/src/lib.rs:375]: What does this change do? 👀

Comment (xxchan) [reply src/utils/runtime/src/lib.rs:375]: Oh I might be wrong. It seems to only affect `fmt` layer.

Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs:375]: Have we tested that we really need this extra step? Calling `init` on the subscriber should already help us to init the `tracing_log` adaptor.

https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/util.rs.html#58-74

Comment (Xuanwo) [reply src/utils/runtime/src/lib.rs:375]: > Calling `init` on the subscriber should already help us to init the `tracing_log` adaptor.

Seems right, we only need to enable `tracing-log` feature.

Comment (wcy-fdu) [reply src/utils/runtime/src/lib.rs:375]: Thanks, I see. `tracing-log` is enabled by default, so we should not do anything here.

**Pull request merged**